### PR TITLE
test storage_proof

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/114?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #113** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/113?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #114** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/114?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/114?utm_source=stack-comment).

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 1]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+            match ext.as_backend().storage($key) {
+                Ok(Some(_)) => (),
+                _ => panic!("There must be some storage changing"),
+            }
+        };
+
+        ([non_present] $key:expr) => {
+            match ext.as_backend().storage($key) {
+                Ok(None) => (),
+                _ => panic!("There must be nothing storage changing"),
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:410]:
```suggestion
                _ => panic!("There must be no storage changing"),
```

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 2]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+fn test_no_commitment_to_storage() {
+    let mut ext = new_test_ext();
+
+    macro_rules! execute_and_commit {
+        ($action:expr) => {
+            ext.execute_with(|| {
+                $action;
+            });
+            ext.commit_all()
+                .expect("Impossible having open transactions");
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:395]:
```suggestion
                .expect("Impossible to have open transactions");
```

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 3]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+        };
+    }
+
+    execute_and_commit!(assert_ok!(Blobs::submit_blob(
+        RuntimeOrigin::signed(alice()),
+        0,
+        get_blob(1024)
+    )));
+
+    assert_key!([present] & TotalBlobSize::<Test>::hashed_key());
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:421]: I find the use of macros here harms readability

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 4]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
         );
     });
 }
+
+#[test]
+fn test_no_commitment_to_storage() {
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:386]:
```suggestion
fn test_no_commitment_to_storage_after_finalization() {
```

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 5]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+
+    execute_and_commit!(Blobs::on_finalize(System::block_number()));
+
+    assert_key!([non_present] & TotalBlobSize::<Test>::hashed_key());
+    assert_key!([non_present] & TotalBlobs::<Test>::hashed_key());
+    assert_key!([non_present] & BlobList::<Test>::hashed_key());
+}
+
+#[test]
+fn test_storage_proof() {
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:433]:
```suggestion
fn test_storage_proof_does_not_contain_ephemeral_keys() {
```

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 6]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+    });
+
+    let memory_bd = proof.into_memory_db();
+    let root = ext.backend.root();
+    let trie =
+        sp_trie::TrieDBBuilder::<LayoutV1<sp_core::Blake2Hasher>>::new(&memory_bd, root).build();
+
+    macro_rules! assert_key {
+        ([present] $key:expr) => {
+            assert!(trie.contains($key).expect("Impossible get value from Trie"));
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:469]:
```suggestion
            assert!(trie.contains($key).expect("trie must contain key"));
```

File: sugondat-chain/pallets/blobs/src/tests.rs [hunk 7]
```diff
@@ -377,3 +381,101 @@ fn test_pre_dispatch_max_total_blobs_size_reached() {
...
+    let (_res, proof) = ext.execute_and_prove(|| {
+        execute_submit_blob!();
+    });
+
+    let memory_bd = proof.into_memory_db();
+    let root = ext.backend.root();
+    let trie =
+        sp_trie::TrieDBBuilder::<LayoutV1<sp_core::Blake2Hasher>>::new(&memory_bd, root).build();
+
+    macro_rules! assert_key {
```
Comment (rphmeier) [inline sugondat-chain/pallets/blobs/src/tests.rs:467]: macro for 1 line substitutions is overkill, can you remove?

**Review by rphmeier — APPROVED** [review]
Looks good except for a few nits - merge when fixed!

File: sugondat-chain/pallets/blobs/src/tests.rs (hunk 7 above)
Comment (gabriele-0201) [reply sugondat-chain/pallets/blobs/src/tests.rs:467]: For sure. Initially, I was also expecting the presence of events in the proof. However, when I removed the check, I didn't update all the code properly. I will do it now

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
### Merge activity

* **Dec 15, 11:39 AM**: @@rphmeier started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/114).

**Pull request merged**